ocs_ci/ocs/platform_nodes.py

```diff
@@ -818,6 +875,205 @@ def _prepare_node(self, node_id):
+        pv = 'ParameterValue'
+
+        param_list.append({pk: 'Index', pv: str(index)})
+        param_list.append({pk: 'InfrastructureName', pv: self.infra_id})
+        param_list.append({pk: 'RhcosAmi', pv: self.worker_image_id})
+        param_list.append(
+            {
+                pk: 'IgnitionLocation', pv: self.worker_ignition_location
+            }
+        )
+        param_list.append({pk: 'Subnet', pv: self.worker_subnet})
+        param_list.append(
+            {
+                pk: 'WorkerSecurityGroupId',
+                pv: self.worker_security_group[0].get('GroupId')
+            }
+        )
+        param_list.append(
+            {
+                pk: 'WorkerInstanceProfileName', pv: self.worker_instance_profile
+            }
+        )
+        param_list.append(
+            {
+                pk: 'WorkerInstanceType', pv: conf['worker_instance_type']
+            }
+        )
+
+        return param_list
+
+    def add_cert_to_template(self, worker_template_path):
+        """
+        Add cert to worker template
+
+        Args:
+            worker_template_path (str): Path where template file is located
+
+        """
+        worker_ignition_path = os.path.join(
+            self.cluster_path,
+            constants.WORKER_IGN
+        )
+        cert = self.get_cert_content(worker_ignition_path)
+        self.update_template_with_cert(worker_template_path, cert)
+
+    def update_template_with_cert(self, worker_template_path, cert):
+        """
+        Update the template file with cert provided
+
+        Args:
+            worker_template_path (str): template file path
+            cert (str): Certificate body
+
+        """
+        search_str = "ABC...xYz=="
+        temp = "/tmp/worker_temp.yaml"
+        with open(worker_template_path, "r") as fp:
+            orig_content = fp.read()
+            logger.info("=====ORIGINAL=====")
+            logger.info(orig_content)
+            final_content = re.sub(
+                r'{}'.format(search_str),
+                r'{}'.format(cert),
+                orig_content
+            )
+        with open(temp, 'w') as wfp:
+            logger.info(final_content)
+            wfp.write(final_content)
+        os.rename(temp, worker_template_path)
+
+    def get_cert_content(self, worker_ignition_path):
+        """
+        Get the certificate content from worker ignition file
+
+        Args:
+            worker_ignition_path (str): Path of the worker ignition file
+
+        Returns:
+            formatted_cert (str): certificate content
+
+        """
+        assert os.path.exists(worker_ignition_path)
+        with open(worker_ignition_path, "r") as fp:
+            content = json.loads(fp.read())
+            cert_content = (
+                content.get
+                ('ignition').get
+                ('security').get
+                ('tls').get
+                ('certificateAuthorities')[0].get
+                ('source')
+            )
```

Comment:
This looks weird. 

Rather use something like:
```python
    cert_content = content.get('ignition').get('security').get(
        'tls'
    ).get('certificateAuthorities')[0].get('source')
```

Or split it to some few variables  like 
```python
tls_data = content.get('ignition').get('security').get('tls')
cert_content = tls_data.get(...
```